fiona/collection.py

```diff
@@ -221,6 +221,87 @@ def crs_wkt(self):
             self._crs_wkt = self.session.get_crs_wkt()
         return self._crs_wkt
 
+    def tags(self, ns=None):
+        """Returns a dict containing copies of the dataset or layers's
+        tags. Tags are pairs of key and value strings. Tags belong to
+        namespaces.  The standard namespaces are: default (None) and
+        'IMAGE_STRUCTURE'.  Applications can create their own additional
+        namespaces.
+
+        Parameters
+        ----------
+        ns: str, optional
+            Can be used to select a namespace other than the default.
+
+        Returns
+        -------
+        dict
+        """
+        if self.session:
+            return self.session.tags(ns=ns)
+        return None
+
+    def get_tag_item(self, key, ns=None):
+        """Returns tag item value
+
+        Parameters
+        ----------
+        key: str
+            The key for the metadata item to fetch.
+        ns: str, optional
+            Used to select a namespace other than the default.
+
+        Returns
+        -------
+        str
+        """
+        if self.session:
+            return self.session.get_tag_item(key=key, ns=ns)
+        return None
+
+    def set_tags(self, tags, ns=None):
+        """Writes a dict containing the dataset or layers's tags.
+        Tags are pairs of key and value strings. Tags belong to
+        namespaces.  The standard namespaces are: default (None) and
+        'IMAGE_STRUCTURE'.  Applications can create their own additional
+        namespaces.
+
+        Parameters
+        ----------
+        tags: dict
+            The dict of metadata items to set.
+        ns: str, optional
+            Used to select a namespace other than the default.
+
+        Returns
+        -------
+        int
+        """
+        if isinstance(self.session, WritingSession):
+            return self.session.set_tags(tags, ns=ns)
+        raise RuntimeError("Unable to set tags as not in writing mode.")
```

Comment:
@snowman2 here I think we should begin with 

```
if not isinstance(self.session, WritingSession):
    raise ...
```

and then I think we should consider a custom error. If we open a Python (3) file to read and then call its write method we get an io.UnsupportedOperation. We should add something like that to fiona.

Reply:
Have a preference for the exception name?